statsd/uds.go

```diff
@@ -43,56 +45,23 @@ func (w *udsWriter) GetTransportName() string {
 	}
 }
 
-// retryOnWriteErr returns true if we should retry writing after a write error
-func (w *udsWriter) retryOnWriteErr(err error, stream bool) bool {
-	// Never retry when using unixgram (to preserve the historical behavior)
-	if !stream {
-		return false
-	}
-	// Otherwise we retry on timeout because we might have written a partial packet
-	if networkError, ok := err.(net.Error); ok && networkError.Timeout() {
+func (w *udsWriter) shouldCloseConnection(err error, partialWrite bool) bool {
+	if err != nil && partialWrite {
+		// We can't recover from a partial write
 		return true
 	}
-	return false
-}
-
-func (w *udsWriter) shouldCloseConnection(err error) bool {
 	if err, isNetworkErr := err.(net.Error); err != nil && (!isNetworkErr || !err.Timeout()) {
 		// Statsd server disconnected, retry connecting at next packet
 		return true
 	}
 	return false
 }
 
-// writeFull writes the whole data to the UDS connection
-func (w *udsWriter) writeFull(data []byte, stopIfNoneWritten bool, stream bool) (int, error) {
```

Comment:
This doesn't seem to be related to connection timeout. Can this change be split in a separate PR?

Reply:
Like in the Java PR the `connectionTimeout` is used instead of the `writeTimeout` to reset the connection if a partial write timeouts.

Do you still think we should add this in a separate PR?